### ISSUE TITLE
Gradle 2.14 break

### ISSUE THREAD

**Issue by ShreyashPromact** [x2on/gradle-hockeyapp-plugin#84]
Hi, I have two different flavor in my app. 1. betatester and 2. hockey
Now, I want to upload hockey flavor build on my hockey account. So I have run the command as follow:

`gradlew uploadHockeyReleaseToHockeyApp`

But that is failed with error message **INSTANCE**

I am not getting where is the exact issue. Please check below Failed message and let me now where might be the issue for that.

`
 :app:zipalignHockeyRelease                 
:app:assembleHockeyRelease                 
:app:uploadHockeyReleaseToHockeyApp                 
App file: D:\AndroidStudioWorkspace\CroziApp\app\build\outputs\apk\app-hockey-release.apk
Mapping file: D:\AndroidStudioWorkspace\CroziApp\app\build\outputs\mapping\hockey\release\mapping.txt
:app:uploadHockeyReleaseToHockeyApp FAILED                        
               
FAILURE: Build failed with an exception.
               
* What went wrong:
Execution failed for task ':app:uploadHockeyReleaseToHockeyApp'.
> INSTANCE     
               
* Try:         
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
               
BUILD FAILED   
               
Total time: 48.746 secs
               
`

Thanks.

Sincerely,
Shreyash

**Comment by x2on** [comment]
Could you please run with `--debug` to get more details

**Comment by DavidAudrain** [comment]
Hi,

I do have the same issue on my side.
Run with --stacktrace, the following exception has been logged:

```
Caused by: java.lang.NoClassDefFoundError: org/gradle/internal/logging/progress/ProgressLoggerFactory
	at de.felixschulze.gradle.HockeyAppUploadTask.uploadFilesToHockeyApp(HockeyAppUploadTask.groovy:156)
	at de.felixschulze.gradle.HockeyAppUploadTask$uploadFilesToHockeyApp.callCurrent(Unknown Source)
	at de.felixschulze.gradle.HockeyAppUploadTask.upload(HockeyAppUploadTask.groovy:150)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 68 more
Caused by: java.lang.ClassNotFoundException: org.gradle.internal.logging.progress.ProgressLoggerFactory
	... 79 more
```
Regards,
David

**Comment by DavidAudrain** [comment]
Sounds like ProgressLoggerFactory is not usable from plugins anymore. See https://github.com/elastic/elasticsearch/issues/18935

**Comment by DavidAudrain** [comment]
I was using gradle 2.10. I am trying with 2.14.

**Comment by DavidAudrain** [comment]
With 2.14, I am getting a different issue:

```
Caused by: java.lang.NoSuchFieldError: INSTANCE
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.<clinit>(SSLConnectionSocketFactory.java:144)
        at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:955)
        at org.apache.http.impl.client.HttpClientBuilder$build$1.call(Unknown Source)
        at de.felixschulze.gradle.HockeyAppUploadTask.uploadFilesToHockeyApp(HockeyAppUploadTask.groovy:176)
        at de.felixschulze.gradle.HockeyAppUploadTask$uploadFilesToHockeyApp.callCurrent(Unknown Source)
        at de.felixschulze.gradle.HockeyAppUploadTask.upload(HockeyAppUploadTask.groovy:150)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)

```

Java is 1.8

```
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)
```

**Comment by x2on** [comment]
Hm ok i think the only way is to completly remove the ProgressLogger from the plugin, or?

**Comment by mlilienberg** [comment]
I have almost the same issue.

`Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory`

I am using
java version "1.8.0_91"
gradle version "2.14.1"

**Comment by stephenroberts82** [comment]
Since moving from 2.10 to 2.14 I am seeing:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app-android:uploadDevReleaseToHockeyApp'.
> No service of type ProgressLoggerFactory available in ProjectScopeServices.

**Comment by touchbee** [comment]
@x2on Any plans on fixing the ProgressLogger issue? Maybe removing the ProgressLogger for the time being?

**Comment by stephenroberts82** [comment]
Are there any plans to fix this? For more than a month the plugin has been unusable for me. It's just to know if I need to find an alternative solution. Thanks.

**Comment by x2on** [comment]
Im currently working on a fix.

**Comment by ShreyashPromact** [comment]
@x2on 

Is that issue resolved?

**Comment by x2on** [comment]
Yes it's fixed in the 3.6 release.
I tested it with gradle 2.14.1 and it worked fine.
